### ISSUE TITLE
QueueRanked Error missing value where TRUE/FALSE needed

### ISSUE THREAD

**Issue by anaks3** [randyzwitch/RSiteCatalyst#240]
Hello,

I know this has been a topic that has been discussed in length, however, I have not been able to find a solution that would work for me yet.

I've tried the solution that was proposed in #201 but unfortunately that did not help either.

My command is:

raw_data_with_products <- QueueTrended("IKEADEPROD",
                                        startdate,
                                        enddate,
                                        c("visits", "entries", "singleaccess", "cm2213_5acf70507a877d203885cac8", 
                                          "cm2213_5acf704be395c27893d1d2d1", "cartviews", "orders", "revenue", 
                                          "units", "cm2213_5acf6c7ede89ed8422aa85af", "event3", "totaltimespent"),
                                        elements = c('product', 'mobiledevicetype'),
                                        classification = c("HFB"),
                                        interval = 40,
                                        segment.id = c("557e95e3e4b0d8e408b03fbb","s2213_57f76a0be4b020d65f60fe2e"),
                                        date.granularity = "day",
                                        top = "1000",
                                        max.attempts = 50000)
[1] "Requesting URL attempt #1"
[1] "Requesting URL attempt #2"
[1] "Requesting URL attempt #3"
[1] "Received trended report."
Error in if (!is.null(elements[i, ]$classification) && nchar(elements[i,  : 
  missing value where TRUE/FALSE needed
Calls: QueueTrended ... GetReport -> ParseTrended -> BuildInnerBreakdownsRecursively
Execution halted

Through my testing I was able to narrow it down to the "elements" (elements = c('product', 'mobiledevicetype')). If I only run the script with "product" or "mobiledevicetype" it executes without any problems. As soon as I include more than one element, however, it breaks down.

I've installed the most recent version of the package (1.4.15).

Any advice would be much appreciated.

Thank you in advance.

Angel

**Comment by anaks3** [comment]
One thing that is worth mentioning is that the script runs without any issues when ran manually. The problem occurs once the script is ran with a batch file.

Thanks

**Comment by anaks3** [comment]
I attempted executing the below suggestion from #207, however, it returns 404 error

library(devtools)
install_github("randyzwitch/RSiteCatalyst", ref = "fix_na")

So when running a quick test I can verify the following observations. 
- I can run two elements and no classification or the "product" element and the "HFB" classification. "HFB" is not a relevant classification for "mobiledevicetype".
- The error does not occur when I run the script without the classification.
- The error does not occur when I run the script manually - only when I try to start it via a batch file.

Unfortunately, I need this exact combination of the elements and the classification for my report. I know that excluding the classification and creating segments will do the job as we've tested that in the past but I was trying to see if there is another possibility before scripting everything.

Am I right in assuming that this is the only possible solution at this point or are there other options I can test?

Thanks

**Comment by slin30** [comment]
The fix in #207 was merged in, so the latest version on CRAN includes it-- no need to pull from dev branch. 

If "product" is the first element and "mobiledevicetype" is the second element, then either of these two should work:

```{r}
elements = c("product", "mobiledevicetype"), 
classification = c("HFB", "")
```

or 

```{r}
elements = c("product", "mobiledevicetype"), 
classification = c("HFB")
```

Only if the element order was reversed would you need to `""` pad the classifications arg. So it's strange that you are getting an error, period, and even more so that it's only via batch. The latter leads me to believe there's something odd going on with the batch script-- is there any possibility that the batch file might be pointing to an older version of the package?

Other than that, the only thing I can suggest is adding the blank (`""`) as in the first code block example to see if that (for some strange reason) helps.

**Comment by anaks3** [comment]
I agree, the issue is very odd. I will review the R setup that we have as none of the solutions that we've discussed has worked for me.

Many thanks for your help.

Angel
